src/main/java/com/espro/flink/consul/metric/ConsulMetricService.java

```diff
@@ -0,0 +1,113 @@
+package com.espro.flink.consul.metric;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.Metric;
+import org.apache.flink.metrics.SimpleCounter;
+import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.DescriptiveStatisticsHistogram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+/**
+ * This class registers metrics to Flink metrics and then calculates and publishes them.
+ */
+public class ConsulMetricService{
+    private static final Logger LOG = LoggerFactory.getLogger(ConsulMetricService.class);
+
+    private static final int HISTORY_SIZE = 100;
+
+    private final MetricRegistry metricRegistry;
+    private final ConsulMetricGroup consulMetricGroup;
+    private final Counter readRequest;
+    private final Counter writeRequest;
+    private final Histogram readDuration;
+    private final Histogram writeDuration;
+    private final Counter session;
+    private final Counter sessionError;
+    private final Histogram sessionDuration;
+
+    public ConsulMetricService(MetricRegistry metricRegistry, ConsulMetricGroup consulMetricGroup) {
+        this.metricRegistry = metricRegistry;
+        this.consulMetricGroup = consulMetricGroup;
+        this.readRequest = new SimpleCounter();
+        this.writeRequest = new SimpleCounter();
+        this.session = new SimpleCounter();
+        this.sessionError = new SimpleCounter();
+        this.readDuration = new DescriptiveStatisticsHistogram(HISTORY_SIZE);
+        this.writeDuration = new DescriptiveStatisticsHistogram(HISTORY_SIZE);
+        this.sessionDuration = new DescriptiveStatisticsHistogram(HISTORY_SIZE);
+    }
+
+    /**
+     * Register the default Consul metrics to the Flink metric entry point
+     */
+    public void registerDefaultMetrics() {
```

Comment:
I would propose to make this method private and invoke it from the constructor rather than relying on "somebody" to call it.